File: src/api_alchemy/Eth/RequestDTO/requestBaseDTO.cs
using System.Text.Json.Serialization;

namespace api_alchemy.Eth.RequestDTO
{

    public class requestBaseDTO
    {
        public string jsonrpc { get; set; } = string.Empty;
        public string method { get; set; } = string.Empty;

        [JsonPropertyName("params")]
        public object[]? _params { get; set; } = null;
        public int id { get; set; } = 0;
    }

}

File: src/f#/apiClients/rpc/alchemy/DTO/request/requestBaseDTO.fs
module requestBaseDTO

open System.Text.Json.Serialization

type requestBaseDTO = {
    jsonrpc: string
    method: string
    [<JsonPropertyName("params")>]
    _params: obj[]
    id: int
}
with
    static member Default = 
         {
              jsonrpc = "2.0"
              method = "eth_getBlockByNumber"
              _params = [||]
              id = 0
          }

